### PR TITLE
MAINT: Audit CIs using zizmor

### PR DESCRIPTION
**Pull request by larsoner**
Not 100% sure about adding this as a pre-commit hook because at least on latest Ubuntu 24.10 I needed a newer rust than was installed on the system, but it was pretty easy to install. An alternative would be to add it as a [GitHub action](https://woodruffw.github.io/zizmor/usage/#use-in-github-actions).

### CODE REVIEW

**Comment by larsoner** [comment]
Opened https://github.com/woodruffw/zizmor/issues/236 about pre-commit.ci, we'll see what they say...

**Review by drammock — COMMENTED** [review]
I think we need to first decide whether there's value in running locally via pre-commit. If not, then we might as well just run it as a GHA instead of through pre-commit.ci.

I don't like the idea of all contributors (not just maintainers!) needing to install or update Rust just to make a change to MNE. That alone sways me pretty strongly toward GHA / running zizmor only in cloud. The only hesitation I have is that, if someone *did* introduce a security hole into one of our GHAs, it would be nice to catch that hole locally *before* it got pushed into a PR and executed.

**Comment by larsoner** [comment]
Actually they responded to the issue and might try to make it Rust 1.75-compatible, which would work with at least Ubuntu 22.04+. Not sure about macOS or Windows, though.

Could add it to the Style GHA run I suppose

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -80,6 +80,13 @@ repos:
         files: pyproject.toml
         additional_dependencies: ["mne"]
 
+  # zizmor
+  # If you need rust, follow
+  # https://doc.rust-lang.org/beta/book/ch01-01-installation.html
+  - repo: https://github.com/woodruffw/zizmor
+    rev: v0.7.0
```
Comment (graingert) [inline .pre-commit-config.yaml:87]:
```suggestion
  - repo: https://github.com/woodruffw/zizmor-pre-commit
    rev: v0.8.0
```

**Comment by graingert** [comment]
> I don't like the idea of all contributors (not just maintainers!) needing to install or update Rust just to make a change to MNE. That alone sways me pretty strongly toward GHA / running zizmor only in cloud. The only hesitation I have is that, if someone did introduce a security hole into one of our GHAs, it would be nice to catch that hole locally before it got pushed into a PR and executed.

no need to install rust anymore! It's a regular python hook - suspiciously similar to ruff

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -80,6 +80,13 @@ repos:
         files: pyproject.toml
         additional_dependencies: ["mne"]
 
+  # zizmor
+  # If you need rust, follow
+  # https://doc.rust-lang.org/beta/book/ch01-01-installation.html
```
Comment (graingert) [inline .pre-commit-config.yaml:85]:
```suggestion
```

**Review by drammock — APPROVED** [review]

**Comment by drammock** [comment]
thanks @larsoner and @graingert !

**Pull request merged**